### PR TITLE
issue/10 - Add ajax search to products dropdown

### PR DESCRIPTION
**Pull request by dbeja**
Issue #10 

**Housekeeping**

* [x] Issue number has been linked below the description
* [x] Branch name follows the `issue/###` or `release/###` format
* [x] PR title is prefixed with the branch name (issue/### - Title)

**Proposed changes**

* [x] Code has been tested with no errors
* [x] Code meets stated minimum requirements
* [x] Code should (usually) not contain unrelated changes
* [x] Code style (including docs) and organization is generally consistent with the rest of the codebase
* [x] Before and after screenshots have been posted to document visual changes
* [x] Clear steps for testing the PR have been posted (as needed)

**Changes**
* Upgraded select2 script to the latest version
* Products dropdown is now dynamic and searchable via ajax. No limit on the number of products.
* When selected it will load with ajax 100 products
* When the list is scrolled down, it will get more pages of products if it exists (infinite scrolling)
* Any terms entered it will search with ajax on the database

**Before changes**
image

**After changes**
Just 2 minor differences: rounded corners and a bit more padding on the input.
As it seems to look better, I didn't force to be exactly as before.
image

### CODE REVIEW

**Comment by dbeja** [comment]
@DrewAPicture Should I also add Andrew here as a reviewer?

**Comment by DrewAPicture** [comment]
> @DrewAPicture Should I also add Andrew here as a reviewer?

Yes please.

**Review by amdrew — APPROVED** [review]
Seems to work great in my testing. It's nice how when you scroll down the list it loads more and more. I was able to load all 400+ of my EDD products inside the dropdown without issue:

👍 from me

Screen Shot 2021-02-17 at 3 46 32 PM

File: includes/scripts.php [hunk 1]
```diff
@@ -30,13 +30,30 @@ function affwp_affiliate_product_rates_admin_footer_js() {
 		return;
 	}
 
+	$rest_url = rest_url( 'affwp_pr/v1/products' );
```
Comment (DrewAPicture) [inline includes/scripts.php:33]: Need to update the REST route here as well.

**Review by DrewAPicture — CHANGES_REQUESTED** [review]

**Comment by dbeja** [comment]
@DrewAPicture, REST routes updated.

**Review by DrewAPicture — APPROVED** [review]
